enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,933 @@
+  provides the documentation of the installation of a cluster in an existing VPC.
+- Phase 1: Install a cluster in user provided network introducing to the installer a new
+  machine pool named `edge`. The MachineSet manifests will be created for each subnet
+  provided on `platform.aws.subnets` for Local Zones. The Machine will be protected with
+  taints to avoid running the normal platform workloads.
+- Phase 2: The installer should automate the network resource creation when the user
+  provides zone names with the type `local-zone` on the 
+  `compute[?name=="edge"].platform.aws.zones`.
+
+Phases 1 and 2 are covered by this enhancement.
+
+## Motivation
+
+OCP Install Support for AWS Local Zones will address Customer Segments where low latency and
+data locality requirements enforce as deal breaker/show-stoppers for the sales teams engagements.
+
+The resources on Local Zones were designed for applications delivering low latency to
+the end-users, in general, hosted in the data plane nodes. Furthermore, the price of
+cloud services is more expensive, and the services are limited. Finally, mixing the
+ingress router backends from different locations in the Local Zones can impact
+drastically, and directly, the end users' experience (ex: routing requests from users
+from NYC to node hosted in Local Zones in Miami, instead of New York node).
+
+As the applications using edge locations are very specific workloads, cluster admins
+want to have available the compute nodes on the board without mixing cluster components
+on those pools, and would like to specify which workloads will run for each location to
+optimize the cloud resources. For that reason, the nodes on the edge pool have taints to
+`NoSchedule`.
+
+The installer supports day 2 operation in Local Zones subnets following these conditions:
+- the VPC must have enough free CIDR blocks to create subnets in Local Zones
+- the Local Zones group must be activated manually
+- the subnets must be created manually
+- the subnet must be associated with a route table
+- the instance type offers must be checked, and the type is chosen for each Local Zone
+- the MachineSets manifests must be created manually for each zone Local
+- preferably, the node should not receive regular cluster workloads, mainly ingress. due the
+  unsupported network load balancers, latency, and costs.
+
+Presently, the installer does not support Local Zones subnets on day zero operations
+for those reasons:
+- the Local Zones do not support network load balancers (NLB or CLB), used by
+  ingress deployed on the `worker` pool.
+- the Local Zones do not support Nat Gateway, used as a gateway on the default IPI
+  deployment for each subnet and availability zone.
+- the default instance type for compute nodes used on IPI, `m6i.xlarge`, is not
+  available on Local Zones
+- the default EBS type for IPI, `gp3`, is not available in the most Local Zones locations
```

Comment:
cluster-storage-operator uses `gp3` as the default storage class. Since most Local Zones can't use such a volume, is there an expecation that OCP is smart enough to use `gp2` as the default for PVs? If so, cluster-storage-operator needs to be changed to deploy gp2 as the default (that's easy) when local zones are detected (that could be hard, is such info available in any API object?)

Reply:
@jsafrane, thanks for pointing it out. Could you please review if this sentence is correct?
- The default StorageClass will be used only when the PV is required. The PV will be used on Local Zones nodes only when users choose to deploy workloads there - due to the taints on those nodes. The users should create tolerations on their deployment spec to run workloads on the edge on day-2, in that steps they should define the `storageClass`=`gp2-csi` when the specific Local Zone location does not support gp3.

Reply:
ack, that sounds good. User is responsible for using the right storage class when a workload is targeting node in a local zone.

Reply:
mentioning this discussion to the [specific EBS section](https://github.com/openshift/enhancements/pull/1232/commits/8b96adf7661e73f0b8b76e9c2e9c5185093df361)